apps/speed.c

```diff
@@ -2977,6 +3061,107 @@ int speed_main(int argc, char **argv)
                 ecdh_doit[testnum] = 0;
         }
     }
+
+    for (testnum = 0; testnum < EdDSA_NUM; testnum++) {
+        int st = 1;
+        EVP_PKEY *ed_pkey = NULL;
+        EVP_PKEY_CTX *ed_pctx = NULL;
+
+        if (!eddsa_doit[testnum])
+            continue;           /* Ignore Curve */
+        for (i = 0; i < loopargs_len; i++) {
+            loopargs[i].eddsa_ctx[testnum] = EVP_MD_CTX_new();
+            if (loopargs[i].eddsa_ctx[testnum] == NULL) {
+                st = 0;
+                break;
+            }
+
+            if ((ed_pctx = EVP_PKEY_CTX_new_id(test_ed_curves[testnum].nid, NULL))
+                    == NULL
+                || !EVP_PKEY_keygen_init(ed_pctx)
+                || !EVP_PKEY_keygen(ed_pctx, &ed_pkey)) {
+                st = 0;
+                break;
+            }
+
+            EVP_DigestSignInit(loopargs[i].eddsa_ctx[testnum], NULL, NULL,
```

Comment:
Also, doing the init outside of the loop works for current EdDSA just because there is no actual message digest, but I think it is conceptually wrong as for example the code as it is written now wouldn't support Prehash-EdDSA or any other signature scheme through EVP.

Reply:
Sorry for the delay of this, working on other stuff these days. I fixed the return value check, but I deferred to put the 'Init" inside the loop at current stage...